Signal/src/ViewControllers/SafetyNumberConfirmationAlert.swift

```diff
@@ -0,0 +1,119 @@
+
+        let displayName: String = {
+            if let signalAccount = contactsManager.signalAccountMap[unconfirmedIdentity.recipientId] {
+                return contactsManager.displayName(for: signalAccount)
+            } else {
+                return contactsManager.displayName(forPhoneIdentifier: unconfirmedIdentity.recipientId)
+            }
+        }()
+
+        let titleFormat = NSLocalizedString("CONFIRM_SENDING_TO_CHANGED_IDENTITY_TITLE_FORMAT",
+                                            comment: "Action sheet title presented when a users's SN have recently changed. Embeds {{contact's name or phone number}}")
+        let title = String(format: titleFormat, displayName)
+
+        let bodyFormat = NSLocalizedString("CONFIRM_SENDING_TO_CHANGED_IDENTITY_BODY_FORMAT",
+                                           comment: "Action sheet body presented when a user's SN have recently changed. Embeds {{contact's name or phone nubmer}}")
+        let body = String(format: bodyFormat, displayName)
+
+        let actionSheetController = UIAlertController(title: title, message:body, preferredStyle: .actionSheet)
+
+        let confirmAction = UIAlertAction(title: confirmationText, style: .default) { _ in
+            Logger.info("\(self.TAG) Confirmed identity: \(unconfirmedIdentity)")
+            OWSDispatch.sessionStoreQueue().async {
+                self.storageManager.saveRemoteIdentity(unconfirmedIdentity.identityKey,
+                                                       recipientId: unconfirmedIdentity.recipientId,
+                                                       approvedForBlockingUse: true,
+                                                       approvedForNonBlockingUse: true)
+                DispatchQueue.main.async {
+                    completion(true)
+                }
+            }
+        }
+        actionSheetController.addAction(confirmAction)
+
+        let showSafetyNumberAction = UIAlertAction(title: NSLocalizedString("VERIFY_PRIVACY", comment: "Action sheet item"), style: .default) { _ in
+            Logger.info("\(self.TAG) Opted to show Safety Number for identity: \(unconfirmedIdentity)")
+
+            self.presentSafetyNumberViewController(fromViewController: fromViewController,
+                                                   theirIdentityKey: unconfirmedIdentity.identityKey,
+                                                   theirRecipientId: unconfirmedIdentity.recipientId,
+                                                   theirDisplayName: displayName,
+                                                   completion: { completion(false) })
+
+        }
+        actionSheetController.addAction(showSafetyNumberAction)
+
+        let dismissAction = UIAlertAction(title: NSLocalizedString("TXT_CANCEL_TITLE", comment: "generic cancel text"), style: .cancel)
+        actionSheetController.addAction(dismissAction)
+
+        fromViewController.present(actionSheetController, animated: true)
```

Comment:
If you use `UIApplication.shared.frontmostViewController?.present(...)` you can remove a parameter. As you like.